### ISSUE TITLE
[Bug] MetadataAndPayload getByHash() throws NoSuchElementException

### ISSUE THREAD

**Issue by ivohuisjes** [ConsenSys/IPFS-Store#58]
When I use the new method: public MetadataAndPayload getByHash(String indexName, String hash)
The backend is throwing: java.util.NoSuchElementException: null

The reason is, I believe, a fault at line 268 of RestIPFSStoreWrapperImpl.java
`Order order = pageable.getSort().iterator().next();`

Here you assume there is a next element, while there is not

Stacktrace:
java.util.NoSuchElementException: null
	at java.util.AbstractList$Itr.next(AbstractList.java:364) ~[na:1.8.0_181]
	at java.util.Collections$UnmodifiableCollection$1.next(Collections.java:1042) ~[na:1.8.0_181]
	at net.consensys.tools.ipfs.ipfsstore.client.java.wrapper.impl.RestIPFSStoreWrapperImpl.search(RestIPFSStoreWrapperImpl.java:268) ~[ipfs-store-client-java-v0.1.17.jar:0.1.17]
	at net.consensys.tools.ipfs.ipfsstore.client.java.IPFSStore.getMetadataByHash(IPFSStore.java:357) ~[ipfs-store-client-java-v0.1.17.jar:0.1.17]
	at net.consensys.tools.ipfs.ipfsstore.client.java.IPFSStore.getByHash(IPFSStore.java:319) ~[ipfs-store-client-java-v0.1.17.jar:0.1.17]
	at nl.enshore.hackathon.backend.service.FileService.getMetadataAndPayload(FileService.java:250) ~[classes/:na]

I am using v0.1.17

**Comment by ivohuisjes** [comment]
Note: I get the same exception, when using your suggested code from here: https://github.com/ConsenSys/IPFS-Store/issues/51#issuecomment-440314622

**Comment by ivohuisjes** [comment]
This is the code I am using for now (that works):

```
public MetadataAndPayload getByHash(String hash) throws IPFSStoreException {
        Query query = Query.newQuery().equals("__hash", hash);
        Sort sort = Sort.by(Sort.Direction.ASC, "title");
        Pageable pageable = PageRequest.of(0, 1, sort);
        Page<MetadataAndPayload> result = ipfsStore.searchAndFetch("documents", query, pageable);
        return result.getContent().stream().findFirst().orElse(null);
    }
```

**Comment by gjeanmart** [comment]
Should be fixed now on 0.2.0

**Comment by ivohuisjes** [comment]
confirmed: fixed
